levant dispatch fails with panic: runtime error: invalid memory address or nil pointer dereference

levant dispatch fails with panic: runtime error: invalid memory address or nil pointer dereference

While trying to dispatch the following parameterized job

```
job "dispatch-test" {
  region      = "global"
  datacenters = ["dc1"]
  type        = "batch"

  parameterized {
    payload       = "forbidden"
    meta_required = ["dir_name"]
  }

  group "create-dir" {

    task "create-dir" {
      driver = "raw_exec"
      template {
        data = <<EOH
db_dir=/opt/data/${NOMAD_META_dir_name}
echo creating $db_dir
mkdir $db_dir
EOH
        destination = "local/run.sh"
      }

      config {
        command = "sh"
        args = ["local/run.sh"]
      }
    }
  }
}
```

using the command

```
levant dispatch -address=nomad.test -meta dir_name=test_db_2 dispatch-test 
```

The following error is received.

```
2020-01-24T03:58:54Z |INFO| levant/dispatch: triggering dispatch against job dispatch-test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x791746]

goroutine 1 [running]:
github.com/jrasell/levant/levant.(*levantDeployment).dispatch(0xc0001c7bc8, 0x7ffd2cb78c73, 0xd, 0xc000191140, 0x0, 0x0, 0x0, 0xff9c6a778701c201)
        /home/travis/gopath/src/github.com/jrasell/levant/levant/dispatch.go:56 +0x266
github.com/jrasell/levant/levant.TriggerDispatch(0x7ffd2cb78c73, 0xd, 0xc000191140, 0x0, 0x0, 0x0, 0x7ffd2cb78c41, 0x18, 0x2)
        /home/travis/gopath/src/github.com/jrasell/levant/levant/dispatch.go:24 +0x17c
github.com/jrasell/levant/command.(*DispatchCommand).Run(0xc00000dbe0, 0xc00001e200, 0x4, 0x4, 0xc00000d8a0)
        /home/travis/gopath/src/github.com/jrasell/levant/command/dispatch.go:116 +0x725
github.com/mitchellh/cli.(*CLI).Run(0xc0000b7e00, 0xc0000b7e00, 0x7, 0xc00000d960)
        /home/travis/gopath/pkg/mod/github.com/mitchellh/cli@v0.0.0-20170908181043-65fcae5817c8/cli.go:255 +0x1da
main.RunCustom(0xc00001e1f0, 0x5, 0x5, 0xc000190ea0, 0x40574f)
        /home/travis/gopath/src/github.com/jrasell/levant/main.go:62 +0x330
main.Run(0xc00001e1f0, 0x5, 0x5, 0xc00007e058)
        /home/travis/gopath/src/github.com/jrasell/levant/main.go:30 +0x56
main.main()
        /home/travis/gopath/src/github.com/jrasell/levant/main.go:24 +0x15e
```

**Output of `levant version`:**

```
Levant v0.2.9
Date: 2019-12-27T09:43:30Z
Commit: 0f1913ea3d77584ddd0696da429dcf05572a73a0
Branch: 0.2.9
State: 0.2.9
Summary: 0f1913ea3d77584ddd0696da429dcf05572a73a0
```

**Output of `consul version`:**

```
(paste your output here)
```

**Output of `nomad version`:**

```
(paste your output here)
```

**Additional environment details:**

**Debug log outputs from Levant:**

<!--
If the log fragment is particularly long, please paste it into a gist or
similar and paste the link here.
-->

--- a/levant/dispatch.go
+++ b/levant/dispatch.go
@@ -2,6 +2,7 @@ package levant
 
 import (
 	"github.com/hashicorp/levant/client"
+	"github.com/hashicorp/levant/levant/structs"
 	nomad "github.com/hashicorp/nomad/api"
 	"github.com/rs/zerolog/log"
 )
@@ -53,9 +54,14 @@ func (l *levantDeployment) dispatch(job string, metaMap map[string]string, paylo
 
 	// In order to correctly run the jobStatusChecker we need to correctly
 	// assign the dispatched job ID/Name based on the invoked job.
-	l.config.Template.Job = &nomad.Job{}
-	l.config.Template.Job.ID = &eval.DispatchedJobID
-	l.config.Template.Job.Name = &eval.DispatchedJobID
+	l.config = &DeployConfig{
+		Template: &structs.TemplateConfig{
+			Job: &nomad.Job{
+				ID:   &eval.DispatchedJobID,
+				Name: &eval.DispatchedJobID,
+			},
+		},
+	}
 
 	// Perform the evaluation inspection to ensure to check for any possible
 	// errors in triggering the dispatch job.

